Wyam.Modules.Xmp/Xmp.cs

```diff
@@ -0,0 +1,353 @@
+﻿using MetadataExtractor;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Text.RegularExpressions;
+using System.Threading.Tasks;
+using Wyam.Common.Documents;
+using Wyam.Common.Modules;
+using Wyam.Common.Pipelines;
+using MetadataExtractor.Formats.Xmp;
+using XmpCore;
+using System.Globalization;
+
+namespace Wyam.Modules.Xmp
+{
+
+    /// <summary>
+    /// This Module Reads specified Xmp Metadata from the Inputs
+    /// and writes it in the Metadata of the Document.  
+    /// </summary>
+    public class Xmp : IModule
+    {
+        private readonly bool _skipElementOnMissingData;
+        private readonly bool _errorOnDoubleKeys;
+        private readonly bool _delocalizing;
+        private readonly bool _flatting;
+        private readonly List<XmpSearchEntry> toSearch = new List<XmpSearchEntry>();
+        private readonly Dictionary<string, string> namespaceAlias = new Dictionary<string, string>();
+
+        /// <summary>
+        /// 
+        /// </summary>
+        /// <param name="skipElementOnMissingMandatoryData">If Mandatory Data is missing this element will not leave this module.</param>
+        /// <param name="errorsOnDoubleKeys">If true, and the Metadata of the input document would written twice by this Module it gives an error. If false the later specified Metadata overrides the first.</param>
+        /// <param name="delocalizing">If multiple elements with diferent laguges are present, the local languge will be used to choose the correct element.</param>
+        /// <param name="flatten">If an array has only one Element it is reduced too this element.</param>
+        public Xmp(bool skipElementOnMissingMandatoryData = false, bool errorsOnDoubleKeys = true, bool delocalizing = true, bool flatten = true)
+        {
+            _skipElementOnMissingData = skipElementOnMissingMandatoryData;
+            _errorOnDoubleKeys = errorsOnDoubleKeys;
+            _delocalizing = delocalizing;
+            _flatting = flatten;
+
+            namespaceAlias["dc"] = "http://purl.org/dc/elements/1.1/";
+            namespaceAlias["xmpRights"] = "http://ns.adobe.com/xap/1.0/rights/";
+            namespaceAlias["cc"] = "http://creativecommons.org/ns#";
+            namespaceAlias["xmp"] = "http://ns.adobe.com/xap/1.0/";
+            namespaceAlias["xml"] = "http://www.w3.org/XML/1998/namespace";
```

Comment:
If these are all always used as initial values, can we move them up to the declaration of `namespaceAlias` as initializer syntax to declutter the constructor? I.e.,:

```C#
private readonly Dictionary<string, string> namespaceAlias =
    new Dictionary<string, string>()
    {
        {"dc", "http://purl.org/dc/elements/1.1/"},
        {"xmpRights", "http://ns.adobe.com/xap/1.0/rights/"},
        // Etc.
    };
```